### ISSUE TITLE
Latex build fails with newer latex

### ISSUE THREAD

**Issue by oscarbenjamin** [sympy/sympy#20326]
The latex build uses this for the `\LaTeX`macro:
https://github.com/sympy/sympy/blob/8df74827994b808a7aa4c8162564d78545e8d2d0/doc/src/conf.py#L189-L190
I'm finding that it breaks on newer xelatex though giving errors like:
```
LaTeX Warning: Command \LaTeX invalid in math mode on input line 88.


LaTeX Warning: Command \TeX invalid in math mode on input line 88.

! You can't use `\spacefactor' in math mode.
\@->\spacefactor 
                 \@m {}
l.88 \(\mathrm{\LaTeX}
                      \) printing by running \sphinxcode{\sphinxupquote{init...
Sorry, but I'm not programmed to handle this case;
I'll just pretend that you didn't ask for it.
If you're in the wrong mode, you might be able to
return to the right one by typing `I}' or `I$' or `I\par'.

Missing character: There is no è in font cmex10!
[1

]
```
The xelatex version is
```
3.14159265-2.6-0.999991 (TeX Live 2019/Debian) (preloaded format=xelatex 2020.10.23)  23 OCT 2020 20:12
```
On another machine I have an older version that has no trouble compiling the latex docs:
```
XeTeX 3.14159265-2.6-0.99998 (TeX Live 2017)
```
I'm not much of an expert on tex but I can see a simple way to fix this by using a new macro like maybe `\newcommand{\mLaTeX}{\text{\LaTeX}}`. That would mean that we needed to use `\mLaTeX` instead of `\LaTeX` in the docs. I guess the current approach is designed to preserve the use of the `\LaTeX` macro but I'm not sure I quite understand how it works and it seems to be broken now.

If anyone has any suggestions let me know. Otherwise I'll go with `\mLaTeX`.

**Comment by asmeurer** [comment]
We already seem to be using \mathrm{\LaTeX} everywhere. Will it work to instead use \text{\LaTeX}?

**Comment by oscarbenjamin** [comment]
As far as I can tell `\text` works and it seems more appropriate to me since we aren't really using this in equations.

**Comment by oscarbenjamin** [comment]
Actually using `\mLaTeX` doesn't work in the html docs because mathjax doesn't see the `\newcommand`.

**Comment by oscarbenjamin** [comment]
This seems to do it for both mathjax and latex provided it's early in the preamble:
```
% Define version of \LaTeX that is usable in math mode
\let\OldLaTeX\LaTeX
\renewcommand{\LaTeX}{\text{\OldLaTeX}}
```

**Comment by oscarbenjamin** [comment]
This is fixed in #20443 which is for the 1.7 branch but it should also be fixed for the master branch.

**Comment by oscarbenjamin** [comment]
This can be closed once #20443 and #20458 are merged
